server/market/orderrouter.go

```diff
@@ -518,10 +399,112 @@ func (r *OrderRouter) handleMarket(user account.AccountID, msg *msgjson.Message)
 		msgID: msg.ID,
 	}
 
+	return r.processTrade(oRecord, tunnel, assets, market.Coins, sell, 0, market.RedeemSig, market.Serialize())
+}
+
+// processTrade checks that the trade is valid and submits it to the market.
+func (r *OrderRouter) processTrade(oRecord *orderRecord, tunnel MarketTunnel, assets *assetSet,
+	coins []*msgjson.Coin, sell bool, rate uint64, redeemSig *msgjson.RedeemSig, sigMsg []byte) *msgjson.Error {
+
+	fundingAsset := assets.funding
+	user := oRecord.order.User()
+	trade := oRecord.order.Trade()
+
+	// If the receiving asset is account-based, we need to check that they can
+	// cover fees for the redemption, since they can't be subtracted from the
+	// received amount.
+	receivingBalancer, isToAccount := assets.receiving.Backend.(asset.AccountBalancer)
+	if isToAccount {
+		if redeemSig == nil {
+			log.Errorf("user %s did not include a RedeemSig for received asset %s", user, assets.receiving.Symbol)
```

Comment:
Are all of the new logs here really error level? I suggest warn, or info, as they are not really a server problem.

Reply:
I'm even thinking DBG for these, but we can keep INF for now I guess.  Anything that a user can do at will with little cost shouldn't be too chatty in the logs.